docs/Compiler-Hardening-Guides/Compiler-Options-Hardening-Guide-for-C-and-C++.md

```diff
@@ -76,9 +76,15 @@ The recommendations in Table 1 and Table 2 are primarily applicable to compiling
 
 For historical reasons, the GCC compiler and Binutils upstream projects do not enable optimization or security hardening options by default. While some aspects of the default options can be changed when building GCC and Binutils from source, the defaults used in the toolchains shipped with GNU/Linux distributions vary. Distributions may also ship multiple versions of toolchains with different defaults. Consequently, developers need to pay attention to compiler and linker options flags, and manage them according to their need of optimization, level of warning and error detection, and security hardening of the project.
 
-You can typically find the default flags used by GCC on your system by examining the output of the `gcc -v` command. Examining the `Configured with:` section, in the `gcc -v` output, can be useful for understanding how GCC is set up on your system and what options were chosen by the distribution's maintainers during the build process. It can also be useful for diagnosing issues with GCC or for troubleshooting problems that may arise during the compilation of software.
+You can typically find the default flags used by GCC on your system by examining the output of the `gcc -v` command. Examining the `Configured with:` section, in the `gcc -v` output, can be useful for understanding how GCC is set up on your system and what options were chosen by the distribution's maintainers during the build process. It can also be useful for diagnosing issues with GCC or for troubleshooting problems that may arise during the compilation of software. For example, issues caused by interactions between order-sensitive flags.
```

Comment:
Should include the corresponding flag for Clang (and possible the Microsoft compiler?)

Possible also the relevant variant for debugging issues with macros (`gcc -O2 -dM -E - < /dev/null`)